src/main/java/httpserver/HttpRequestParser.java

```diff
@@ -0,0 +1,21 @@
+package httpserver;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class HttpRequestParser {
+    public static HttpRequest fromString(String rawRequest){
+        List<String> chunkedRawRequest = Arrays.asList(rawRequest.split("\r\n"));
+        String[] statusLine = chunkedRawRequest.get(0).split(" ");
```

Comment:
Do you need to use an array literal.

Reply:
I'm looking into matchers, so maybe not!